src/libstd/sys_common/util.rs

```diff
@@ -22,7 +22,14 @@ pub fn min_stack() -> usize {
         n => return n - 1,
     }
     let amt = env::var("RUST_MIN_STACK").ok().and_then(|s| s.parse().ok());
+
+    #[cfg(not(target_os = "l4re"))]
     let amt = amt.unwrap_or(2 * 1024 * 1024);
+
+    // L4Re is not able to handle more than 1MB.
+    #[cfg(target_os = "l4re")]
+    let amt = amt.unwrap_or(1 * 1024 * 1024);
```

Comment:
Unfortunately `tidy` disallows you from writing this using `#[cfg]` or `cfg!()`. See https://github.com/rust-lang/rust/blob/master/src/tools/tidy/src/pal.rs for detail.

```
[00:03:46] tidy error: /checkout/src/libstd/sys_common/util.rs:26: platform-specific cfg: cfg(not(target_os = "l4re"))
[00:03:46] tidy error: /checkout/src/libstd/sys_common/util.rs:30: platform-specific cfg: cfg(target_os = "l4re")
```